src/vm/eventpipeconfiguration.h

```diff
@@ -89,16 +89,17 @@ class EventPipeConfiguration
     }
 
 private:
-    // Helper function used to generate a "EventPipeSession ID" (bitmask).
-    EventPipeSessionID GenerateSessionId() const
+    // Helper function used to locate a free index in the range 0 - EventPipe::MaxNumberOfSessions
+    // Returns EventPipe::MaxNumberOfSessions if there are no free indexes
+    unsigned int GenerateSessionIndex() const
     {
         LIMITED_METHOD_CONTRACT;
-
+        _ASSERTE(EventPipe::MaxNumberOfSessions == 64);
```

Comment:
nit: Prefer compile time error. Both numbers are `const`, this assert could be a `static_assert`.